runtime/interpreter/value.go

```diff
@@ -17180,7 +17198,9 @@ func (v *EphemeralReferenceValue) MeteredString(memoryGauge common.MemoryGauge,
 func (v *EphemeralReferenceValue) StaticType(inter *Interpreter) StaticType {
 	referencedValue := v.ReferencedValue(inter, EmptyLocationRange)
 	if referencedValue == nil {
-		panic(DereferenceError{})
+		panic(DereferenceError{
+			Cause: "the value being referenced has been destroyed or moved",
```

Comment:
In a follow-up PR we can maybe pass along the location range, but that requires passing it to `Value.StaticType()`